src/main/java/org/cactoos/func/RunnableOf.java

```diff
@@ -23,79 +23,61 @@
  */
 package org.cactoos.func;
 
-import java.util.concurrent.Callable;
 import org.cactoos.Func;
 import org.cactoos.Proc;
+import org.cactoos.Scalar;
 import org.cactoos.proc.ProcOf;
-import org.cactoos.proc.UncheckedProc;
+import org.cactoos.scalar.Unchecked;
 
 /**
  * Func as Runnable.
  *
  * <p>There is no thread-safety guarantee.
  *
- * @param <X> Type of input
  * @since 0.12
  */
-public final class RunnableOf<X> implements Runnable {
+public final class RunnableOf implements Runnable {
 
     /**
-     * Original proc.
+     * Runnable.
      */
-    private final Proc<X> proc;
-
-    /**
-     * The input.
-     */
-    private final X input;
-
-    /**
-     * Ctor.
-     * @param proc Encapsulated proc
-     * @since 0.11
-     */
-    public RunnableOf(final Callable<X> proc) {
-        this(new FuncOf<>(proc));
-    }
-
-    /**
-     * Ctor.
-     * @param proc Encapsulated proc
-     */
-    public RunnableOf(final Proc<X> proc) {
-        this(proc, null);
-    }
+    private final Scalar<?> scalar;
```

Comment:
@victornoel  Why is it a `Scalar` and not a `Runnable`?

Reply:
@andreoss Runnable.run does not throw an exception so it makes it awkward to assign a scalar (from the constructor) into a  runnable (while the opposite is acceptable). Now that I think about it, I could have wrapped the scalar into unchecked at the constructor level. I will do that for consistency sake with other similar classes.